common/awsClient.go

```diff
@@ -0,0 +1,102 @@
+// nolint:ireturn
+package common
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+)
+
+type ctxKey string
+
+const AWSServiceContextKey ctxKey = "AWSService"
+
+// ErrRequestAWSMissingService is returned when deep connector implementation doesn't attach
+// AWS service name into the context, therefore the request cannot be constructed for sending.
+var ErrRequestAWSMissingService = errors.New("AWS request is missing Service name, supplied via context")
+
+type AWSClient struct {
+	client *http.Client
+	cfg    aws.Config
+	region string
+}
+
+func NewAWSClient(ctx context.Context, client *http.Client,
```

Comment:
can you add some comments to explain why AWS is special and requires its own client? I know from reading the Slab doc, but it would be good to have it here a comment also .